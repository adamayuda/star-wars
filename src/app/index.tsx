import "./style.sass";

import React, { FC, Suspense, lazy } from "react";

import { Route, RouteProps, Switch } from "react-router-dom";

import { Loading } from "src/components/loading";

/* eslint-disable  @typescript-eslint/no-explicit-any */
interface RouteInterface extends RouteProps {
  import: Promise<any>;
}

const routes: RouteInterface[] = [
  {
    import: import("src/pages/home"),
    path: "/",
    exact: true,
  },
  {
    import: import("src/pages/login"),
    path: "/login",
    exact: true,
  },
];

export const App: FC = () => {
  return (
    <Suspense fallback={<Loading />}>
      <Switch>
        {routes.map((route, index: number) => (
          <Route
            {...route}
            key={`route-${index}`}
            component={lazy(() => route.import)}
          />
        ))}
      </Switch>
    </Suspense>
  );
};
