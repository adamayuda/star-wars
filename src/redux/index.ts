import { applyMiddleware, compose, createStore } from "redux";
import { mainReducer } from "src/redux/reducers";
import thunk from "redux-thunk";

const composeEnhancers =
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const mainStore = createStore(
  mainReducer,
  composeEnhancers(applyMiddleware(thunk)),
);

export const createMainStore = () => {
  return createStore(mainReducer, composeEnhancers(applyMiddleware(thunk)));
};

export type StateInterface = ReturnType<typeof mainStore.getState>;

export type ActionType = "USER_LOGIN";

export interface Action<T> {
  type: ActionType;
  payload: Partial<T>;
}
