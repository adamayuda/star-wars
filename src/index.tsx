import { App } from "src/app";
import { Provider } from "react-redux";

import React from "react";
import ReactDOM from "react-dom";

import { BrowserRouter as Router } from "react-router-dom";

import { mainStore } from "src/redux";

ReactDOM.render(
  <Provider store={mainStore}>
    <Router>
      <App />
    </Router>
  </Provider>,
  document.getElementById("root"),
);
