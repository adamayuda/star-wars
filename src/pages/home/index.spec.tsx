import { cleanup, render, screen, waitFor } from "@testing-library/react";

import { Home } from ".";
import { Provider } from "react-redux";
import React from "react";

import { BrowserRouter as Router } from "react-router-dom";
import { createMainStore } from "src/redux";

let mainStore = createMainStore();

describe("src/pages/home/index.tsx", () => {
  afterEach(() => cleanup());
  beforeEach(() => {
    mainStore = createMainStore();
  });

  test("Render Home page", async () => {
    const renderedHome = render(
      <Provider store={mainStore}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    );

    expect(renderedHome.container).toMatchSnapshot();
  });
});
