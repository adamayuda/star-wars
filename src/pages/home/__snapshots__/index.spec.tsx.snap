// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/home/index.tsx Render Home page 1`] = `
<div>
  <nav>
    <div
      class="container"
    >
      <div
        class="navGrid"
      >
        <img
          alt="LinkedTeam"
          src="test-file-stub"
        />
        <img
          alt="LinkedTeam"
          src="test-file-stub"
        />
        <img
          alt="LinkedTeam"
          src="test-file-stub"
        />
      </div>
    </div>
  </nav>
  <div
    class="navBottom"
  >
    <div
      class="navGrid"
    >
      <img
        alt="LinkedTeam"
        src="test-file-stub"
      />
      <img
        alt="LinkedTeam"
        src="test-file-stub"
      />
      <img
        alt="LinkedTeam"
        src="test-file-stub"
      />
      <img
        alt="LinkedTeam"
        src="test-file-stub"
      />
      <img
        alt="LinkedTeam"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    class="home"
  >
    <div
      class="container"
    >
      <div
        class="posts"
      />
    </div>
  </div>
</div>
`;
